javascript/node/operational-vs-programmer-errors/handling-operational-errors.md

```diff
@@ -0,0 +1,98 @@
+# Handling Operational Errors
+author: mihaiberq
+
+levels:
+  - beginner
+  - basic
+  - medium
+
+type: normal
+
+category: how to
+
+tags:
+  - introduction
+  - workout
+
+links:
+  - '[Node JS and Errors](https://www.joyent.com/node-js/production/design/errors)'
+
+---
+## Content
+
+There isn't a single place in the application where errors can pop up, so you should be prepared to handle them wherever you are making a HTTP request, performing I/O operations, forking processes or validation input. You may end up handling the same error at different points of the stack, since those functions can't do more than passing the error.
+
+For most operational errors, the message will most likely point to what is left to do. For example, trying to access a file that is not yet created will *return* (async) or *throw* (sync) an `ENOENT: no such file or directory` error.
+
+While we're at it, let's focus on the difference between asynchronous and synchronous functions. Consider the following snippet:
+```js
+import fs from 'fs';
+
+function readAndParseFile(path,
+  user_callback) {
+    try {
+      fs.read(path, (err, text) => {
```

Comment:
`fs.read` takes a file descriptor, which is an integer, you are likely thinking of `fs.readFile` here.